src/platform/singleton.h

```diff
@@ -33,18 +24,11 @@ class Singleton
         is_destructed = true;
     }
 private:
-    static bool is_destructed;
-
-    static boost::mutex& GetMutex()
-    {
-        static boost::mutex mutex;
-        return mutex;
-    }
+    static volatile bool is_destructed;
```

Comment:
No longer needed, the static member enforced the creation of the mutex in static init phase ( single thread ), so mutexes could be used right away. With c++11 singletons, no mutexes are needed, so this force-creation is not required either.